projects/packages/search/src/class-module-control.php

```diff
@@ -148,23 +148,49 @@ public function deactivate() {
 		 */
 		do_action( 'jetpack_deactivate_module_' . self::JETPACK_SEARCH_MODULE_SLUG );
 
+		$this->disable_instant_search();
+
 		return $success;
 	}
 
+	/**
+	 * Update module status
+	 *
+	 * @param boolean $active - true to activate, false to deactivate.
+	 */
+	public function update_status( $active ) {
+		return $active ? $this->activate() : $this->deactivate();
+	}
+
 	/**
 	 * Disable Instant Search Experience
 	 */
 	public function disable_instant_search() {
+		if ( ! $this->is_instant_search_enabled() ) {
+			return true;
+		}
 		return update_option( self::SEARCH_MODULE_INSTANT_SEARCH_OPTION_KEY, false );
 	}
 
 	/**
 	 * Enable Instant Search Experience
 	 */
 	public function enable_instant_search() {
+		if ( ! $this->is_active() || $this->is_instant_search_enabled() ) {
```

Comment:
Shouldn't this return `false` for `! $this->is_active()` to indicate that the option was not updated? In other words, wouldn't this scenario be equivalent to a 400 error?

Reply:
Good catch here. If the module is not active then we definitely should return false or an error.

Reply:
addressed in 3d57b30851